packages/yt-dlp-ejs/build.sh

```diff
@@ -0,0 +1,23 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/yt-dlp/ejs
+TERMUX_PKG_DESCRIPTION="External JavaScript for yt-dlp supporting many runtimes"
+TERMUX_PKG_LICENSE="Unlicense"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION="0.3.1"
+TERMUX_PKG_SRCURL="https://github.com/yt-dlp/ejs/releases/download/$TERMUX_PKG_VERSION/yt_dlp_ejs-$TERMUX_PKG_VERSION.tar.gz"
+TERMUX_PKG_SHA256=7f2119eb02864800f651fa33825ddfe13d152a1f730fa103d9864f091df24227
+TERMUX_PKG_DEPENDS="deno | nodejs | nodejs-lts, python"
```

Comment:
I think we can make this work for 32 Bit architectures with a conditional.

```suggestion
TERMUX_PKG_DEPENDS="deno | nodejs | nodejs-lts, python"
if (( TERMUX_ARCH_BITS == 32 )); then
	TERMUX_PKG_DEPENDS="nodejs | nodejs-lts, python"
fi
```

Reply:
also changing `deno | nodejs | nodejs-lts` -> `deno` since node isnt usable unless `--js-runtimes node` flag is passed to yt-dlp and using node as a fallback on 64bit devices doesnt make much sense

users can can install node and use it manually if they want

Reply:
Yes, but they wouldn't be able to get rid of `deno` in that case since the dependency wouldn't be satisfied, so this should stay `deno | nodejs | nodejs-lts` to make that actually possible as an option.

Reply:
It's true that the dependency of Arch Linux's `yt-dlp` package on `deno` was just removed, but this commit doesn't seem to make a lot of sense to me and just occurred recently,

https://gitlab.archlinux.org/archlinux/packaging/packages/yt-dlp/-/commit/944dbe853ef51e04224a1b54c2ef9eb0cd015cf0

Reasons it doesn't seem to make sense:

- Neither Arch Linux's `yt-dlp` nor Arch Linux's `yt-dlp-ejs` now pulls in `deno` at all, so clean installs appear as though they might break in Arch Linux
- The linked issue in Arch Linux is slightly incoherent, reading "deno should not be an optional dependency... deno should an optional dependency", so it's not quite clear whether it's been properly fixed

It seems to me like Arch Linux might still be in the process of figuring out and settling on a dependency structure, so if they are to be used as an example, I suggest waiting a while to see what happens.

Reply:
Fwiw, `yt-dlp-ejs` in Arch depends on Deno.
https://gitlab.archlinux.org/archlinux/packaging/packages/yt-dlp-ejs/-/blob/main/PKGBUILD

And `yt-dlp-ejs` is an optdepend of `yt-dlp`.
Arch doesn't make the Suggests/Recommends distinction.

Reply:
> is there a specific reason that you believe the deno | nodejs | nodejs-lts dependency shouldn't be in yt-dlp as upstream recommends instead of yt-dlp-ejs?

one reason i can come up with is: in the future yt-dlp will be adding some javascript directly into their main repository instead of splitting it into different repo like they did with ejs

but that hasnt happened yet and doing it either way should be fine for now

Reply:
moved deno/node dependency to `yt-dlp`